Project/Assets/ML-Agents/Examples/Walker/Scripts/WalkerAgent.cs

```diff
@@ -9,14 +9,26 @@
 
 public class WalkerAgent : Agent
 {
-    public float maximumWalkingSpeed = 999; //The max walk velocity magnitude an agent will be rewarded for
-    Vector3 m_WalkDir; //Direction to the target
-//    Quaternion m_WalkDirLookRot; //Will hold the rotation to our target
+    [Header("Walk Speed")] [Range(0, 10)] public float walkingSpeed = 10; //The walking speed to try and achieve
+    float m_maxWalkingSpeed = 10; //The max walking speed
 
-    [Header("Target To Walk Towards")] [Space(10)]
-    public TargetController target; //Target the agent will walk towards.
+    //Should the agent sample a new goal velocity each episode?
+    //If true, walkSpeed will be randomly set between zero and m_maxWalkingSpeed in OnEpisodeBegin() 
+    //If false, the goal velocity will be walkingSpeed
+    public bool randomizeWalkSpeedEachEpisode;
 
-    [Header("Body Parts")] [Space(10)] public Transform hips;
+    public enum WalkDirectionMethod
+    {
+        UseWorldDirection,
```

Comment:
added comments and explanation

Reply:
You cannot have `Method` in this enum as method means something in CS.
Rename UseWorldDirection to Manual and UseTarget to Follow Target.

It would be a lot better to write an editor script to selectively hide parts of the inspector...

Reply:
removed WalkDirectionMethod logic. uses previous target system again.